packages/labs/task/src/task.ts

```diff
@@ -3,441 +3,16 @@
-    } catch (e) {
-      errored = true;
-      error = e;
-    }
-    // If this is the most recent task call, process this value.
-    if (this._callId === key) {
-      if (result === initialState) {
-        this.status = TaskStatus.INITIAL;
-      } else {
-        if (errored === false) {
-          try {
-            this._onComplete?.(result as R);
-          } catch {
-            // Ignore user errors from onComplete.
-          }
-          this.status = TaskStatus.COMPLETE;
-          this._resolveTaskComplete?.(result as R);
-        } else {
-          try {
-            this._onError?.(error);
-          } catch {
-            // Ignore user errors from onError.
-          }
-          this.status = TaskStatus.ERROR;
-          this._rejectTaskComplete?.(error);
-        }
-        this._value = result as R;
-        this._error = error;
-      }
-      // Request an update with the final value.
-      this._host.requestUpdate();
-    }
-  }
-
-  /**
-   * Aborts the currently pending task run by aborting the AbortSignal
-   * passed to the task function.
-   *
-   * Aborting a task does nothing if the task is not running: ie, in the
-   * complete, error, or initial states.
-   *
-   * Aborting a task does not automatically cancel the task function. The task
-   * function must be written to accept the AbortSignal and either forward it
-   * to other APIs like `fetch()`, or handle cancellation manually by using
-   * [`signal.throwIfAborted()`]{@link https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted}
-   * or the
-   * [`abort`]{@link https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/abort_event}
-   * event.
-   *
-   * @param reason The reason for aborting. Passed to
-   *     `AbortController.abort()`.
-   */
-  abort(reason?: unknown) {
-    if (this.status === TaskStatus.PENDING) {
-      this._abortController?.abort(reason);
-    }
-  }
-
-  get value() {
-    return this._value;
-  }
-
-  get error() {
-    return this._error;
-  }
-
-  render<T extends StatusRenderer<R>>(renderer: T) {
-    switch (this.status) {
-      case TaskStatus.INITIAL:
-        return renderer.initial?.() as MaybeReturnType<T['initial']>;
-      case TaskStatus.PENDING:
-        return renderer.pending?.() as MaybeReturnType<T['pending']>;
-      case TaskStatus.COMPLETE:
-        return renderer.complete?.(this.value!) as MaybeReturnType<
-          T['complete']
-        >;
-      case TaskStatus.ERROR:
-        return renderer.error?.(this.error) as MaybeReturnType<T['error']>;
-      default:
-        throw new Error(`Unexpected status: ${this.status}`);
-    }
-  }
-}
-
-type MaybeReturnType<F> = F extends (...args: never[]) => infer R
-  ? R
-  : undefined;
-
-export const shallowArrayEquals = <T extends ReadonlyArray<unknown>>(
-  oldArgs: T,
-  newArgs: T
-) =>
-  oldArgs === newArgs ||
-  (oldArgs.length === newArgs.length &&
-    oldArgs.every((v, i) => !notEqual(v, newArgs[i])));
+export {
```

Comment:
should we re-export with `@deprecated`? or we leave as is and just mark the package as deprecated in the npm registry?

Reply:
Yeah, good idea!